### PR TITLE
Ready for review Java1

### PR DESCRIPTION
**Pull request by nbalangrid**

### CODE REVIEW

File: src/com/java1/reverse/ReverseString.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    }
+
+    public static String reverseTheString4(String input){
+
+        int noOfChars= input.length();
+
+        char[] charArray = new char[noOfChars];
+
+        for(int i=0;i<noOfChars/2;i++){
```
Comment (vcrisac) [inline src/com/java1/reverse/ReverseString.java:41]: the approach of swapping 2 characters at the same time is fine, however, I noticed that when there in the string are an odd number of characters (including spaces), some of the spaces are just skipped and the unicode of the empty element from the array remains there.

Here's the example of what I mean:
Input: `Hello World` - 11 characters (10 chars + a space).
Expected output: `dlroW olleH`.
Actual output: `dlroWolleH`
image

And also, when we test this method, the assertion fails:
`Exception in thread "main" org.junit.ComparisonFailure: expected:<dlrow[ ]olleh> but was:<dlrow[]olleh>`

So my suggestion here would be to just iterate through the whole string `for(int i=0; i<noOfChars; i++)` and swap the elements by picking one element at a time - leave the 42nd line of code and remove 43rd one.

Comment (vcrisac) [inline src/com/java1/reverse/ReverseString.java:41]: Also, I would suggest implementing several Junit / TestNG tests, to validate that the method returns the expected result when it is given different inputs.

**Comment by vcrisac** [comment]
One more comment here: please create a `.gitignore` file and put in it the `.idea` and `.target` folders and the `*.iml` files, so they would not be committed and pushed to the remote repo.

File: src/com/java1/array/SortedArray.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
+package com.java1.array;
+
+import java.util.StringTokenizer;
+
+public class SortedArray {
+
+    int nrElementeArray;
```
[the remark this answers is not in the document]
Comment (nbalangrid) [reply src/com/java1/array/SortedArray.java]: Modified the access for 
private int nrElementeArray;
private int [] array;

File: src/com/java1/array/SortedArray.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+    public void printArray(){
+        System.out.println("Nr elem = " + nrElementeArray);
+        for(int i=0;i<nrElementeArray;i++){
+            System.out.print(array[i]+" ");
+        }
+    }
+
+    public boolean isInAscOrder(){
+        boolean isSorted = true;
```
[the remark this answers is not in the document]
Comment (nbalangrid) [reply src/com/java1/array/SortedArray.java:28]: the initial assumption is that the array is sorted and if any of the consecutive components are not in order the isSorted parameter became false and the array is not sorted

File: src/com/java1/array/SortedArray.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        if(nrElementeArray == 1) return true;
+        for(int i=1 ; i<nrElementeArray;i++){
+            if(array[i] < array[i-1]){
+                isSorted = false;
+            }
+        }
+        if(isSorted){
```
[the remark this answers is not in the document]
Comment (nbalangrid) [reply src/com/java1/array/SortedArray.java]: modify implemented

File: src/com/java1/interfaces/AccountServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
+package com.java1.interfaces;
+
+ class AccountServiceImpl implements AccountService {
+
+     Account[] accounts;
```
[the remark this answers is not in the document]
Comment (nbalangrid) [reply src/com/java1/interfaces/AccountServiceImpl.java]: made the field private

File: src/com/java1/reverse/ReverseString.java (hunk 1 above)
Comment (nbalangrid) [reply src/com/java1/reverse/ReverseString.java]: Implemented the. modifications suggested

Comment (nbalangrid) [reply src/com/java1/reverse/ReverseString.java]: added in the Main class the asserts : assertEquals()

File: src/com/java1/reverse/Main.java [hunk 6]
```diff
@@ -2,19 +2,30 @@
...
+        System.out.println("Output 1: "+output1);
+        String output2 = ReverseString.reverseTheString(s);
+        System.out.println("Output 2: "+output2);
+        Assert.assertEquals(output2,output1);
+        String output3 = ReverseString.reverseTheString(s);
+        System.out.println("Output 3: "+output3);
+        Assert.assertEquals(output3,output1);
+        String output4 = ReverseString.reverseTheString(s);
+        System.out.println("Output 4: "+output4);
+        Assert.assertEquals(output4,output1);
```
Comment (vcrisac) [inline src/com/java1/reverse/Main.java:27]: would you like to implement some tests (TestNG or JUnit) and do the assertions there? 
if yes, then they should be implemented as follows:
`src/com/test/java1/reverse/ReverseStringTest.java` - this is the correct structure for test classes, and in the class should be implemented the testing method.

File: .gitignore [hunk 7]
```diff
@@ -2,6 +2,9 @@
 out/
 !**/src/main/**/out/
 !**/src/test/**/out/
+.idea
+.target
+*.iml
```
[the remark this answers is not in the document]
Comment (nbalangrid) [reply .gitignore:9]: updated the .gitignore file to exclude the libraries folder

File: src/com/java1/reverse/ReverseStringTest.java [hunk 8]
```diff
@@ -0,0 +1,38 @@
+package com.java1.reverse;
```
Comment (vcrisac) [inline src/com/java1/reverse/ReverseStringTest.java:1]: great job!

File: src/com/java1/reverse/Main.java [hunk 9]
```diff
@@ -2,19 +2,30 @@
 
 
 import java.util.Scanner;
+import org.testng.Assert;
+import org.testng.annotations.Test;
 
 public class Main {
 
+
```
[the remark this answers is not in the document]
Comment (nbalangrid) [reply src/com/java1/reverse/Main.java]: done

File: src/com/java1/reverse/Main.java (hunk 6 above)
Comment (nbalangrid) [reply src/com/java1/reverse/Main.java]: implemented ReverseStringTest class

**Comment by vcrisac** [comment]
For some reason I still can see the `.idea`, `.iml` and `libraries` folders and files in the pr. Please make sure they are removed from the PR.
One more thing here - before merging into master, please squash the commits into 1 single.

**Comment by vcrisac** [comment]
> For some reason I still can see the `.idea`, `.iml` and `libraries` folders and files in the pr. Please make sure they are removed from the PR. One more thing here - before merging into master, please squash the commits into 1 single.

done!

**Pull request merged**